Dockerfile

```
@@ -2,16 +2,19 @@ FROM python:3-slim AS builder
 
 RUN mkdir /src
 COPY . /src/
+ENV VIRTUAL_ENV=/opt/venv
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 RUN pip install --no-cache-dir --upgrade pip setuptools wheel \
     # Install build tools to compile dependencies that don't have prebuilt wheels
     && apt update && apt install -y git build-essential \
     && cd /src \
-    && pip install --user --no-cache-dir .[colorama,d]
+    && pip install --no-cache-dir .[colorama,d]
 
 FROM python:3-slim
 
 # copy only Python packages to limit the image size
-COPY --from=builder /root/.local /root/.local
-ENV PATH=/root/.local/bin:$PATH
+COPY --from=builder /opt/venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
 
 CMD ["black"]
```

```suggestion
CMD ["/opt/venv/bin/black"]
```
Lets do this so it's more evident to people looking that we're in a venv and this is the black binary you should expect to use ...

---

Got it, adding this as well